plugins/CoreAdminHome/API.php

```diff
@@ -299,20 +299,29 @@ public function archiveReports($idSite, $period, $date, $segment = false, $plugi
 
         $period = Factory::build($period, $date);
         $site = new Site($idSite);
+        $segmentObj = new Segment(
+            $segment,
+            [$idSite],
+            $period->getDateTimeStart()->setTimezone($site->getTimezone()),
+            $period->getDateTimeEnd()->setTimezone($site->getTimezone())
+        );
         $parameters = new ArchiveProcessor\Parameters(
             $site,
             $period,
-            new Segment(
-                $segment,
-                [$idSite],
-                $period->getDateTimeStart()->setTimezone($site->getTimezone()),
-                $period->getDateTimeEnd()->setTimezone($site->getTimezone())
-            )
+            $segmentObj
         );
         if ($report) {
             $parameters->setArchiveOnlyReport($report);
         }
 
+        Piwik::postEvent('CoreAdminHome.archiveReports.start', [
```

Comment:
All events should have a description/usage example. And/or be flagged with `@internal`. Otherwise it will be listed in the official [developer documentation](https://developer.matomo.org/api-reference/events) without the usual information.